sys-boot/refind/refind-0.11.4.ebuild

```diff
@@ -0,0 +1,177 @@
+	!gnuefi? ( >=sys-boot/udk-2018-r1 )"
+
+DOCS=(README.txt)
+PATCHES=(
+	"${FILESDIR}/makefile.patch"
+	"${FILESDIR}/fsmakefile.patch"
+)
+UDK_WORKSPACE="${T}/udk"
+
+pkg_pretend() {
+	if use custom-cflags; then
+		ewarn
+		ewarn "You have enabled building with USE=custom-cflags. Be aware that"
+		ewarn "using this can result in EFI binaries that fail to run and may"
+		ewarn "fail to build at all. This is strongly advised against by upstream."
+		ewarn
+		ewarn "See https://bugs.gentoo.org/598587#c3 for more information"
+		ewarn
+	fi
+}
+
+pkg_setup() {
+	if use x86; then
+		export EFIARCH=ia32
+		export BUILDARCH=ia32
+	elif use amd64; then
+		export EFIARCH=x64
+		export BUILDARCH=x86_64
+	fi
+}
+
+src_prepare() {
+	default
+
+	# bug 598647 - PIE not supported
+	sed -e '/^CFLAGS/s:$: -fno-PIE:' -i Make.common || die
+
+	# Prepare UDK workspace
+	if ! use gnuefi; then
+		mkdir "${UDK_WORKSPACE}" || die
+		ln -s "${EPREFIX}/usr/lib/udk/"{Mde,IntelFramework}{,Module}Pkg \
+			"${UDK_WORKSPACE}" || die "Could not link UDK files"
+
+		# HFS filesystem is not working with UDK
+		# See https://sourceforge.net/p/refind/discussion/general/thread/4e203f0a10/
+		sed -e "/RefindPkg\/filesystems\/hfs\.inf/d" -i RefindPkg.dsc || die
+		if use hfs; then
+			ewarn "hfs use flag ignored because not working with UDK"
+		fi
+	fi
+}
+
+src_configure() {
+	if ! use gnuefi; then
+		# Use the side effect of the script which will create configuration files
+		(. udk-workspace "${UDK_WORKSPACE}" || die)
+		sed -e "s:^#\?\s*\(MAX_CONCURRENT_THREAD_NUMBER\s*=\).*$:\1 $(makeopts_jobs):" \
+			-i "${UDK_WORKSPACE}/Conf/target.txt" || die "Failed to configure target file"
+		sed -e "s:\(_\(CC\|ASM\|PP\|VFRPP\|ASLCC\|ASLPP\|DLINK\)_PATH\s*=\).*$:\1 $(tc-getCC):" \
+			-e "s:\(_ASLDLINK_PATH\s*=\).*$:\1 $(tc-getLD):" \
+			-e "s:\(_OBJCOPY_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+			-e "s:\(_RC_PATH\s*=\).*$:\1 $(tc-getOBJCOPY):" \
+			-e "s:\(_SLINK_PATH\s*=\).*$:\1 $(tc-getAR):" \
+			-i "${UDK_WORKSPACE}/Conf/tools_def.txt" \
+			|| die "Failed to prepare tools definition file"
+	fi
+}
+
+src_compile() {
+	# Update fs targets depending on uses
+	local fs fs_names=()
+	for fs in ${FS_USE}; do
+		fs=${fs#+}
+		if use "${fs}"; then
+			fs_names+=(${fs})
+		fi
+	done
+	use gnuefi && fs_names=("${fs_names[@]/%/_gnuefi}")
+
+	# Prepare flags
+	local make_flags=(
+		ARCH="${BUILDARCH}"
+		CC="$(tc-getCC)"
+		AS="$(tc-getAS)"
+		LD="$(tc-getLD)"
+		AR="$(tc-getAR)"
+		RANLIB="$(tc-getRANLIB)"
+		OBJCOPY="$(tc-getOBJCOPY)"
+		GNUEFILIB="/usr/$(get_libdir)"
+		EFILIB="/usr/$(get_libdir)"
+		EFICRT0="/usr/$(get_libdir)"
+		EDK2BASE="${UDK_WORKSPACE}"
+		EDK2_DRIVER_BASENAMES="${fs_names[@]}"
+		FILESYSTEMS="${fs_names[@]}"
+		FILESYSTEMS_GNUEFI="${fs_names[@]}"
+	)
+	if use custom-cflags; then
+		make_flags=(CFLAGS="${CFLAGS}" "${make_flags[@]}")
+	fi
+
+	emake "${make_flags[@]}" all_$(usex gnuefi gnuefi edk2)
+}
+
+src_install() {
+	exeinto "/usr/lib/${P}"
```

Comment:
is there actually a reason why to use packagename-version `${P}` instead of just the packagename `${PN}` ?

Reply:
Well, the reason is that, in the future, we may want to slot the package, because versions are not always fully ascending compatible. Note that I create a symlink from `${PN}` to `${P}` anyway.

Reply:
Sorry, I mixed up the `udk` package and the `refind` one. No, there is no good reason for this one. I will update this.